```
make -C /nix/store/9lph03skr2gb4412shxla4ij9m6wspmc-linux-4.19.288-dev/lib/modules/4.19.288/build M=/build/source modules
  CC [M]  /build/source/fanout.o
/build/source/fanout.c:28:12: fatal error: linux/device/class.h: No such file or directory
   28 | #  include <linux/device/class.h>
      |            ^~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
make[4]: *** [/nix/store/9lph03skr2gb4412shxla4ij9m6wspmc-linux-4.19.288-dev/lib/modules/4.19.288/source/scripts/Makefile.build:310: /build/source/fanout.o] Error 1
make[3]: *** [/nix/store/9lph03skr2gb4412shxla4ij9m6wspmc-linux-4.19.288-dev/lib/modules/4.19.288/source/Makefile:1562: _module_/build/source] Error 2
make[2]: *** [Makefile:153: sub-make] Error 2
make[1]: *** [Makefile:24: __sub-make] Error 2
make: *** [Makefile:8: all] Error 2
```

It seems to be failing with this on some of the kernels

--- a/pkgs/os-specific/linux/fanout/default.nix
+++ b/pkgs/os-specific/linux/fanout/default.nix
@@ -1,0 +1,37 @@
+{ lib, stdenv, fetchFromGitHub, kernel, kmod }:
+
+stdenv.mkDerivation rec {
+  pname = "fanout";
+  version = "unstable-2022-10-17-${kernel.version}";
+
+  src = fetchFromGitHub {
+    owner = "bob-linuxtoys";
+    repo = "fanout";
+    rev = "69b1cc69bf425d1a5f83b4e84d41272f1caa0144";
+    hash = "sha256-Q19c88KDFu0A6MejZgKYei9J2693EjRkKtR9hcRcHa0=";
+  };
+
+  preBuild = ''
+    substituteInPlace Makefile --replace "modules_install" "INSTALL_MOD_PATH=$out modules_install"
+  '';
+
+  patches = [
+    ./remove_auto_mknod.patch
+  ];
+
+  hardeningDisable = [ "format" "pic" ];
+
+  nativeBuildInputs = [ kmod ] ++ kernel.moduleBuildDependencies;
+
+  makeFlags = kernel.makeFlags ++ [
+    "KERNELDIR=${kernel.dev}/lib/modules/${kernel.modDirVersion}/build"
+  ];
+
+  meta = with lib; {
+    description = "Kernel-based publish-subscribe system";
+    homepage = "https://github.com/bob-linuxtoys/fanout";
+    license = licenses.gpl2Only;
+    maintainers = with maintainers; [ therishidesai ];
+    platforms = platforms.linux;
+  };
+}

--- a/pkgs/os-specific/linux/fanout/remove_auto_mknod.patch
+++ b/pkgs/os-specific/linux/fanout/remove_auto_mknod.patch
@@ -1,0 +1,13 @@
+diff --git a/fanout.c b/fanout.c
+index f5d2a55..87125f4 100644
+--- a/fanout.c
++++ b/fanout.c
+@@ -13,7 +13,7 @@
+ /* Comment out to forgo the creation of /dev entries
+  * The companion udev rules 'fanout.rules' sets the special file mode
+  */
+-#define DEV_MKNOD
++// #define DEV_MKNOD
+ 
+ #include <linux/kernel.h>
+ #include <linux/module.h>

--- a/pkgs/top-level/linux-kernels.nix
+++ b/pkgs/top-level/linux-kernels.nix
@@ -349,6 +349,8 @@ in {
 
     evdi = callPackage ../os-specific/linux/evdi { };
 
+    fanout = callPackage ../os-specific/linux/fanout { };
+
     fwts-efi-runtime = callPackage ../os-specific/linux/fwts/module.nix { };
 
     gcadapter-oc-kmod = callPackage ../os-specific/linux/gcadapter-oc-kmod { };

